code/game/turfs/simulated/wall_attacks.dm

```diff
@@ -360,3 +360,31 @@
 	else if(!istype(W,/obj/item/rfd/construction) && !istype(W, /obj/item/reagent_containers))
 		return attack_hand(user)
 
+/turf/simulated/shuttle/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(!destructible)
+		return
+	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+	if (!user.)
+		to_chat(user, "<span class='warning'>You don't have the dexterity to do this!</span>")
+		return
+
+	//get the user's location
+	if(!istype(user.loc, /turf))	return	//can't do this stuff whilst inside objects and such
+
+	if(damage && W.iswelder())

```

Comment:
Might want to add a message saying that there's no damage to repair if `damage` is 0.